src/ApiFactory.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\CdtClient;
+
+use OpenEuropa\CdtClient\Contract\ApiFactoryInterface;
+use OpenEuropa\CdtClient\Contract\RestInterface;
+use OpenEuropa\CdtClient\Endpoint\EndpointBase;
+use OpenEuropa\CdtClient\Endpoint\IdentifierEndpoint;
+use OpenEuropa\CdtClient\Endpoint\MainEndpoint;
+use OpenEuropa\CdtClient\Endpoint\ReferenceDataEndpoint;
+use OpenEuropa\CdtClient\Endpoint\RequestsEndpoint;
+use OpenEuropa\CdtClient\Endpoint\StatusEndpoint;
+use OpenEuropa\CdtClient\Endpoint\TokenEndpoint;
+use OpenEuropa\CdtClient\Endpoint\ValidateEndpoint;
+use OpenEuropa\CdtClient\Http\Download;
+use OpenEuropa\CdtClient\Model\Response\Token;
+use OpenEuropa\CdtClient\Traits\ConfigurationAwareTrait;
+
+/**
+ * Class ApiFactory
+ *
+ * Provides a factory for creating API endpoints.
+ *
+ * @see ApiClientInterface
+ */
+class ApiFactory implements ApiFactoryInterface
+{
+    use ConfigurationAwareTrait;
+
+    protected ?Token $token = null;
+
+    /**
+     * @param array<int|string, mixed> $configuration
+     */
+    public function __construct(protected RestInterface $rest, protected array $configuration)
+    {
+    }
+
+    public function setToken(Token $token): ApiFactoryInterface
+    {
+        $this->token = $token;
+        return $this;
+    }
+
+    public function createTokenEndpoint(): TokenEndpoint
+    {
+        return new TokenEndpoint($this->rest, $this->extractConfigValues([
+            'username',
+            'password',
+            'client',
+            'apiBaseUrl',
+        ]));
+    }
+
+    public function createEndpoint(string $class): EndpointBase
+    {
+        if ($class === TokenEndpoint::class) {
```

Comment:
I've added this one to avoid any confusion in the future. Otherwise, PHP will throw an error about invalid configuration, which is misleading.